pkg/autoscaler/scaling/autoscaler.go

```diff
@@ -135,8 +139,8 @@ func (a *autoscaler) Update(deciderSpec *DeciderSpec) error {
 func (a *autoscaler) Scale(ctx context.Context, now time.Time) ScaleResult {
 	logger := logging.FromContext(ctx)
 
-	spec, podCounter := a.currentSpecAndPC()
-	originalReadyPodsCount, err := podCounter.ReadyCount()
+	spec := a.currentSpec()
+	originalReadyPodsCount, err := a.podCounter.ReadyCount()
```

Comment:
Why is it no longer necessary to guide `podCounter` with a RWMutex?

Reply:
since it no longer changes.

Reply:
a vestige of old code.